backend/routes/friends.js

```diff
@@ -5,24 +5,62 @@ var cors = require('cors');
 
 router.use(cors())
 
-router.get('/users/:name', async (req, res) => {
+/**
+ * find user based on username search and filter out people you're already friends with
+ */
+router.get('/users', async (req, res) => {
     try {
+        let sessionQuery = new Parse.Query("_Session").equalTo("sessionToken", req.query.sessionToken)
+        let userId = await sessionQuery.first({useMasterKey : true})
+        userId = userId.attributes.user.id
         let query = new Parse.Query("_User")
-        query.equalTo("username", req.params.name)
+        query.equalTo("username", req.query.name)
+        query.notEqualTo("objectId", userId)
+
+        //to exclude existing friends or already sent friend requests
+        let query1 = new Parse.Query("Friends").equalTo("fromName", req.query.name).equalTo("toUser", userId)
```

Comment:
You should also consider checking first if  req.query.name belongs to a user, before trying to search in the Friends table